ifrs17/Utils/ImportCalculationMethods.ipynb

```diff
@@ -225,9 +225,9 @@
         {
             "cell_type": "code",
             "source": [
-                "public static double[] Interpolate(this double[] cashflowValues, CashFlowPeriodicity periodicity, InterpolationMethod interpolationMethod)",
+                "public static double[] Interpolate(this double[] cashflowValues, string amountType, CashFlowPeriodicity periodicity, InterpolationMethod interpolationMethod)",
                 "\n{ ",
-                "\n    if (periodicity == CashFlowPeriodicity.Monthly)",
+                "\n    if (periodicity == CashFlowPeriodicity.Monthly || (amountType == AmountTypes.ACA || amountType == AmountTypes.AEA)) // TODO remove hardcoding!!",
```

Comment:
as now you can specify in the cashflow file that a row has monthly granularity, you don't need any logic on the amountType or EstimateType.

What you need is in the cashflowImporter: 
1. parse the new column Periodicity
2. when calling Interpolate(parsingStorage.GetCashFlowPeriodicity(dataNode), parsingStorage.GetInterpolationMethod(dataNode)) pass the value read from the column in the Cashflow if present and not null, else pass the value in the parameter (as it is implemented in develop)

Reply:
for nor now I only removed the "hardcoded check on monthly. the above suggestion is not yet implemented.

Reply:
I have now fixed the Cashflow Importer to do the following: 
1) check the presence of 'CashFlowPeriodicity' and 'InterpolationMethod' in the Cashflow, if available, it will use it to "parse" the provided cashflow. Note this is distinct by AmountType and AocStep.
2) If any of the 2 parameters is not available, it will check the 'DataNode' level, namely 'SingleDataNodeParameter' and use for all AmountTypes and AocStep.
3) In case no information or invalid data is provided, the default values willl be used: CashFlowPeriodicity = Monthly and InterpolationMethod = NotApplicable.